src/core/ai.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2018 The boardgame.io Authors
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+/**
+ * AI
+ *
+ * Sets up the necessary information for running bots in a game.
+ *
+ * There are two parts that need to be described:
+ * - What are the next possible moves given a particular game state.
+ * - What is the evaluation function (score) for a particular state.
+ *
+ * A bot might use this information to maximize its score and minimize
+ * other player's scores (i.e. MiniMax and similar algorithms).
+ *
+ * @param {...object} possibleMoves - Function with signature (G, ctx) => moves
+ *                                    that returns possible moves,
+ *
+ *                                    OR
+ *
+ *                                    Object with two keys:
+ *                                    {
+ *                                      ranges:
+ *                                        Object with key as move names
+ *                                        and values with arguments ranges. Each
+ *                                        range can be an object with {min, max, step}
+ *                                        or a list of possible values. Ex:
+ *                                        { clickCell: [{min: 0, max: 8}] }
```

Comment:
What about moves with multiple arguments?

Reply:
This works with moves with multiple arguments. Each argument can have its own range object or array of possible values. See for instance this test with two arguments:
https://github.com/Felizardo/boardgame.io/blob/master/src/core/ai.test.js#L99

Reply:
I think the comment is missing the point here, it should say that it is an array of objects. I will fix that.